STYLEGUIDE.md

```diff
@@ -0,0 +1,212 @@
+# DC/OS CLI Style Guide
+
+## Intro
+The DC/OS CLI is the main tool used by developers and operators to interact with DC/OS cluster and services running on the clusters. The teams developing features and CLI plugins should build the CLI per this style guide so that we offer a consistent and usable experience to users of DC/OS.
+
+## Naming Commands
+* Generally top level commands are single nouns e.g. dcos job, dcos service
+* Top level commands are followed by verbs e.g. dcos job create, dcos cluster rename
+* Command names should always be a single, lowercase word without spaces, hyphens, underscores, or other word delimiters
```

Comment:
We currently have `dcos auth list-providers`, in cases like this `dcos auth providers list` would be overkill as this is the only command and it is not meant to grow. Can we say that commands might contains a hyphen delimiter?

Reply:
sounds good